resources/lib/vrtplayer/streamservice.py

```diff
@@ -204,7 +204,7 @@ def get_stream(self, video, retry=False, api_data=None):
                 drm_stream = (vudrm_token or uplynk)
 
                 # Select streaming protocol
-                if not drm_stream and self._kodi.has_inputstream_adaptive_installed() or drm_stream and self._can_play_drm and self._kodi.get_setting('usedrm') == 'true':
+                if not drm_stream and self._kodi.has_inputstream_adaptive() or drm_stream and self._can_play_drm and self._kodi.get_setting('usedrm') == 'true':
```

Comment:
If I remember it right **and** has precedence over **or** which means that all of the following conditions need to be met:
- not drm_stream
- has_inputstream_adaptive or drm_stream
- can_play_drm
- kodi.get_setting('usedrm') == 'true'

So if **drm_stream is False** the second item only relies on **has_inputstream_adaptive** to be True. So this condition can be simplified to:
```python
if not drm_stream and self._kodi.has_inputstream_adaptive() and self._can_play_drm and self._kodi.get_setting('usedrm') == 'true':
```

Reply:
I prefer to rewrite it like this:
```python
                if not drm_stream and self._kodi.has_inputstream_adaptive():
                    protocol = 'mpeg_dash'
                elif drm_stream and self._can_play_drm and self._kodi.get_setting('usedrm') == 'true':
                    protocol = 'mpeg_dash'
```
Still, I am not sure this is what is intended if we disable the use of inputstream.adaptive.

I am also not sure how MPEG_DASH or HLS related to inputstream.adaptive.

- Can we play both MPEG_DASH and HLS without inputstream.adaptive ?
- Can we play both MPEG_DASH and HLS with inputstream.adaptive ?

Would it make sense to also have an option like:
- Preferred streaming format: HLS / MPEG_DASH

Maybe we need a matrix in the Wiki to

Reply:
1) InputStream Adaptive is indispensable to play MPEG DASH (with or without Widevine DRM protection)
2) Widevine DRM-support (for InputStream Adaptive) is not available on Kodi 17 at all.
3) We can play HLS with Inputstream Adaptive but also with standard Kodi player
4) In Kodi 17 different older versions of InputStream Adaptive are available with less functionality.
5) There are 7(!) different Kodi 17 releases and it's unclear how well InputStream Adaptive works on all these versions. 

So, in streamservice I made choices to limit all possible streaming combinations:
- InputStream Adaptive installed: always play MPEG-DASH
- InputStream Adaptive is missing: always play HLS

This is a simple workflow and it guarantees video and audio will always play on every Kodi version and platform.

Making HLS or MPEG-DASH a user choice makes little sense. The `useinputstreamadaptive` setting basically has the same effect.

Splitting up the if statement for cosmetics is okay (both ` can_play_drm` and `has_inputstream_adaptive` functions listen to the `useinputstreamadaptive` setting.)

Documenting stream compatibility in the wiki won't hurt but is not a priority for me.

Reply:
@mediaminister Thanks for the detailed description. It may be useful to detail this in the code, and maybe even in the settings when we disable InputStream Adaptive (e.g. this forces use of HLS) in the help (which isn't shown *yet*, needs fixing in Kodi).

Reply:
@mediaminister I noticed you have been promoted to *Collaborator* so you can now assign things to me, or ask me to review something. Welcome :-)